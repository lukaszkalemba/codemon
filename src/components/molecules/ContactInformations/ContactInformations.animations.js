const getWrapper = animate => {
  return {
    variants: {
      animate: {
        opacity: 1,
        y: 0,
        transition: { duration: 0.8 },
      },
      initial: { opacity: 0, y: 20 },
    },
    animate,
    initial: 'initial',
  };
};

export default { getWrapper };
