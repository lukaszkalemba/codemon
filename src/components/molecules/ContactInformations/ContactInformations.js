import React, { useEffect } from 'react';
import { useInView } from 'react-intersection-observer';
import { useAnimation } from 'framer-motion';
import animations from './ContactInformations.animations';
import S from './ContactInformations.styles';

const ContactInformations = () => {
  const animation = useAnimation();

  const [wrapperRef, inView] = useInView({
    triggerOnce: true,
    threshold: 0.5,
  });

  useEffect(() => {
    if (inView) {
      animation.start('animate');
    }
  }, [animation, inView]);

  const wrapperAnimations = animations.getWrapper(animation);

  return (
    <S.Wrapper ref={wrapperRef} {...wrapperAnimations}>
      <S.Link href="tel:+48787709856">+48 787 709 856</S.Link>
      <S.Link href="mailto:lukasz.kalemba1@gmail.com">
        lukasz.kalemba1@gmail.com
      </S.Link>
    </S.Wrapper>
  );
};

export default ContactInformations;
