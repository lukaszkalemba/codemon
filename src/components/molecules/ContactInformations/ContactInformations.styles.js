import styled from 'styled-components';
import { motion } from 'framer-motion';

const Wrapper = styled(motion.div)`
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  ${({ theme }) => theme.media.laptop`
    justify-content: center;
  `}
`;

const Link = styled.a`
  font-size: ${({ theme }) => theme.font.size.body.s};
  color: ${({ theme }) => theme.color.white};
  text-decoration: none;
  transition: opacity 400ms ease-in-out;
  padding: 13px 0;
  opacity: 0.8;

  ${({ theme }) => theme.media.laptop`
    padding: 12px 0;
  `}
    :hover {
    opacity: 1;
  }
`;

export default { Wrapper, Link };
