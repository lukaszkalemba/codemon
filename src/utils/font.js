export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
