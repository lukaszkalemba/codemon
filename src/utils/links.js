export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
