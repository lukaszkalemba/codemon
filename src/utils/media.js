export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
