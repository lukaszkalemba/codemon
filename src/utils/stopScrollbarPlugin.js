import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
