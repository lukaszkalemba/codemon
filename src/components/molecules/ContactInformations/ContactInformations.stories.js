import React from 'react';
import ContactInformations from './ContactInformations';

export default {
  title: 'Molecules/ContactInformations',
  component: ContactInformations,
};

const Template = args => <ContactInformations {...args} />;

export const Default = Template.bind({});
Default.args = {};
