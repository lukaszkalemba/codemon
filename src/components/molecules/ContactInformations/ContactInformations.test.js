import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import ContactInformations from './ContactInformations';

describe('<ContactInformations />', () => {
  it('matches snapshot', () => {
    const { container } = render(<ContactInformations />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
