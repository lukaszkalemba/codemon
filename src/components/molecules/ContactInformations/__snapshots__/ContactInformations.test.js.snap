// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactInformations /> matches snapshot 1`] = `
<div
  class="sc-bdnylx hvmfPr"
  style="opacity: 0; transform: translateY(20px) translateZ(0);"
>
  <a
    class="sc-gtssRu ciPAMe"
    href="tel:+48787709856"
  >
    +48 787 709 856
  </a>
  <a
    class="sc-gtssRu ciPAMe"
    href="mailto:lukasz.kalemba1@gmail.com"
  >
    lukasz.kalemba1@gmail.com
  </a>
</div>
`;
